### PR TITLE
[ESPNCricInfo] Add extractor.

### PR DESCRIPTION
**Pull request by Ashish0804**
## Please follow the guide below

- You will be asked some questions, please read them **carefully** and answer honestly
- Put an `x` into all the boxes [ ] relevant to your *pull request* (like that [x])
- Use *Preview* tab to see how your *pull request* will actually look like

---

### Before submitting a *pull request* make sure you have:
- [ ] At least skimmed through [contributing guidelines](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions) including [yt-dlp coding conventions](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#yt-dlp-coding-conventions)
- [ ] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [ ] Checked the code with [flake8](https://pypi.python.org/pypi/flake8)

### In order to be accepted and merged into yt-dlp each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [ ] Bug fix
- [ ] Improvement
- [x] New extractor
- [ ] New feature

---

### Description of your *pull request* and other information

Closes https://github.com/yt-dlp/yt-dlp/issues/1635

### CODE REVIEW

**Comment by Ashish0804** [comment]
@2011 can u test this (i only tested with videos available in India)
also if you can provide of an example with subs and/or with direct video, i can add them aswell.

**Comment by 2011** [comment]
> @2011 can u test this (i only tested with videos available in India) also if you can provide of an example with subs and/or with direct video, i can add them aswell.

Happy to test it, but I will need some instructions on how to obtain the patched version (still a newbie at this). I have always just used the single file releases in the past.

**Comment by Ashish0804** [comment]
if you have pip/python3 installed
Then something like this should work

`python3 -m pip3 install --force-reinstall https://github.com/Ashish0804/yt-dlp/archive/espncricinfo.zip`
may have to change `python3` to just `python`

**Comment by 2011** [comment]
> if you have pip/python3 installed Then something like this should work

Don't have that, but I downloaded the file referenced in the link you gave, and unzipped it. It worked perfectly on a few videos that I have tried so far.

Thanks again.

File: yt_dlp/extractor/espn.py [hunk 1]
```diff
@@ -236,3 +237,44 @@ def _real_extract(self, url):
...
+                formats.append({
+                    'url': item['url'],
+                    'vcodec': 'none',
+                })
+        self._sort_formats(formats)
+        return {
+            'id': id,
+            'title': data_json.get('title'),
+            'description': data_json.get('summary'),
+            'upload_date': unified_strdate(data_json.get('publishedAt') or data_json.get('recordedAt')),
```
Comment (pukkandan) [inline yt_dlp/extractor/espn.py:276]: dict_get

**Review by pukkandan — APPROVED** [review]
assuming you have tested, this can be merged

File: yt_dlp/extractor/espn.py (hunk 1 above)
Comment (Ashish0804) [reply yt_dlp/extractor/espn.py]: done

**Pull request merged**